File: app/src/main/java/com/aditya/calculator/views/SquareButton.java
package com.aditya.calculator.views;

import android.content.Context;
import android.util.AttributeSet;
import android.widget.Button;

public class SquareButton extends android.support.v7.widget.AppCompatButton {

    public SquareButton(Context context) {
        super(context);
    }

    public SquareButton(Context context, AttributeSet attrs) {
        super(context, attrs);
    }

    public SquareButton(Context context, AttributeSet attrs, int defStyle) {
        super(context, attrs, defStyle);
    }

    @Override
    protected void onMeasure(int widthMeasureSpec, int heightMeasureSpec) {
        int measuredWidth = getDefaultSize(getSuggestedMinimumWidth(),
                widthMeasureSpec);
        int measuredHeight = getDefaultSize(getSuggestedMinimumHeight(),
                heightMeasureSpec);
        // Ensure this view is always square.
        int min = Math.min(measuredHeight, measuredWidth);
        setMeasuredDimension(min, min);
    }
}

File: app/src/main/java/com/aditya/calculator/views/SquareButton.kt
package com.aditya.calculator.views

import android.content.Context
import android.util.AttributeSet
import android.view.View
import androidx.appcompat.widget.AppCompatButton
import kotlin.math.min

class SquareButton : AppCompatButton {
    constructor(context: Context?) : super(context)
    constructor(context: Context?, attrs: AttributeSet?) : super(context, attrs)
    constructor(context: Context?, attrs: AttributeSet?, defStyle: Int) : super(context, attrs, defStyle)

    override fun onMeasure(widthMeasureSpec: Int, heightMeasureSpec: Int) {
        val measuredWidth = View.getDefaultSize(suggestedMinimumWidth,
                widthMeasureSpec)
        val measuredHeight = View.getDefaultSize(suggestedMinimumHeight,
                heightMeasureSpec)
        // Ensure this view is always square.
        val min = min(measuredHeight, measuredWidth)
        setMeasuredDimension(min, min)
    }
}